libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp

```diff
@@ -99,13 +102,15 @@ bool ESP8266WiFiAPClass::softAP(const char* ssid, const char* passphrase, int ch
         return false;
     }
 
-    if(!ssid || strlen(ssid) == 0 || strlen(ssid) > 31) {
+    size_t ssid_len = strlen(ssid);
+    if(!ssid || ssid_len == 0 || ssid_len > 32) {
         // fail SSID too long or missing!
         DEBUG_WIFI("[AP] SSID too long or missing!\n");
         return false;
     }
 
-    if(passphrase && strlen(passphrase) > 0 && (strlen(passphrase) > 63 || strlen(passphrase) < 8)) {
+    size_t pass_len = strlen(passphrase);

```

Comment:
passphrase nullptr case is checked after strlen()

Reply:
do you mean to just use
```cpp
if(pass_len > 0 && (pass_len > 63 || pass_len < 8)) {
```
?